### PR TITLE
[BUGFIX] Use paginator in 'SearchController.php' and 'CollectionController.php'

### PR DESCRIPTION
**Pull request by csidirop**
This PR should fix my long standing issue with missing search results raised here https://github.com/kitodo/kitodo-presentation/pull/885#issuecomment-1443367080 and further discussed here #1029.

After a lot of digging, I found the root of the issue. But to be honest I'm not sure if my solution is the best. I just copied existing working code from https://github.com/kitodo/kitodo-presentation/blob/master/Classes/Controller/ListViewController.php#L104 and made it fit into the SearchController and the CollectionController, without understanding if it was missing for a reason. I assume that it was simply overlooked.

Now both ways to set up the search are working (https://github.com/kitodo/kitodo-presentation/issues/1029#issuecomment-1792648218) as well as the collections.

As you can see, both are working:
grafik

### CODE REVIEW

**Comment by csidirop** [comment]
As a side effect, this also seams to have some influence the issue with the gridpage:

prior: 
grafik

after: 
grafik

Unfortunately clicking on the numbers only reloads the first page. I will look further into that asap. But let me know if you sse why this happens!

File: Classes/Controller/CollectionController.php [hunk 1]
```diff
@@ -198,10 +200,26 @@ public function showAction(Collection $collection): void
         $sortableMetadata = $this->metadataRepository->findByIsSortable(true);
 
         // get all documents of given collection
-        $solrResults = $this->documentRepository->findSolrByCollection($collection, $this->settings, $searchParams, $listedMetadata);
+        $solrResults = null;
+        // $numResults = 0;
```
Comment (sebastian-meyer) [inline Classes/Controller/CollectionController.php]: Please remove inactive and debug code.

File: Classes/Controller/CollectionController.php [hunk 2]
```diff
@@ -198,10 +200,26 @@ public function showAction(Collection $collection): void
...
+        if (is_array($searchParams) && !empty($searchParams)) {
+            $solrResults = $this->documentRepository->findSolrByCollection($collection, $this->settings, $searchParams, $listedMetadata);
+            // $numResults = $solrResults->getNumFound();
```
Comment (sebastian-meyer) [inline Classes/Controller/CollectionController.php]: Please remove inactive and debug code.

File: Classes/Controller/CollectionController.php [hunk 3]
```diff
@@ -198,10 +200,26 @@ public function showAction(Collection $collection): void
...
+            $solrPaginator = new SolrPaginator($solrResults, $currentPage, $itemsPerPage);
+            $simplePagination = new SimplePagination($solrPaginator);
+
+            $pagination = $this->buildSimplePagination($simplePagination, $solrPaginator);
+            $this->view->assignMultiple([ 'pagination' => $pagination, 'paginator' => $solrPaginator ]);
+        }
 
         $this->view->assign('viewData', $this->viewData);
         $this->view->assign('documents', $solrResults);
+        // $this->view->assign('numResults', $numResults);
```
Comment (sebastian-meyer) [inline Classes/Controller/CollectionController.php]: Please remove inactive and debug code.

**Review by sebastian-meyer — CHANGES_REQUESTED** [review]

File: Classes/Controller/CollectionController.php (hunk 3 above)
Comment (csidirop) [reply Classes/Controller/CollectionController.php]: Right, I forgot to ask if `numResults` should be set. But it seams not. I removed it

**Review by sebastian-meyer — APPROVED** [review]

**Pull request merged**